### PR TITLE
docs, libobs: Profiler Documentation

### PR DESCRIPTION
**Pull request by Xaymar-Streamlabs**
Adds DoxyGen based documentation to the profiler.

### CODE REVIEW

File: libobs/util/profiler.h [hunk 1]
```diff
@@ -14,44 +14,223 @@ typedef struct profiler_time_entry profiler_time_entry_t;
 /* ------------------------------------------------------------------------- */
 /* Profiling */
 
+/*!
+ * \brief Register a new named root Scope.
+ * 
+ * A named root is used to differentiate between profiling sessions. For
```
Comment (palana) [inline libobs/util/profiler.h]: The actual purpose of this call is to set an expected time between calls (and enable tracking of time between calls); unless you want to track recurring events (such as video rendering) you don't ever need to call this (even when profiling a different thread)

Comment (Xaymar-Streamlabs) [inline libobs/util/profiler.h]: I've changed the text to reflect this now.

**Comment by ONSBalder** [comment]
Can one of the admins verify this patch?

**Comment by jp9000** [comment]
Should be superseded by the sphinx documentation: https://github.com/jp9000/obs-studio/commit/d51e2a019bb6c166731800f25de035b1f99891f3

**Pull request closed without merging**